### PR TITLE
validator/tests: add ANN queries integration test

### PR DESCRIPTION
**Pull request by QuerthDP**
Add test for retuned result from ANN query.
Add test ensuring LIMIT is respected.

Fixes: VECTOR-163

### CODE REVIEW

File: crates/validator/src/tests/ann.rs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+    let correct = rows
+        .iter()
+        .filter(|row| {
+            let pk: i32 = row.columns[0].as_ref().unwrap().as_int().unwrap();
+            pk < 100
+        })
+        .count();
+    info!("Number of matching results: {}", correct);
+    assert!(
+        correct >= 85,
```
Comment (knowack1) [inline crates/validator/src/tests/ann.rs:128]: Why 85%? Could you please explain.

File: crates/validator/src/tests/ann.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+    .await;
+    assert_eq!(rows.len(), 100);
+    let correct = rows
+        .iter()
+        .filter(|row| {
+            let pk: i32 = row.columns[0].as_ref().unwrap().as_int().unwrap();
+            pk < 100
+        })
+        .count();
+    info!("Number of matching results: {}", correct);
```
[the remark this answers is not in the document]
Comment (QuerthDP) [reply crates/validator/src/tests/ann.rs:126]: Not at all. Might remove or move to debug. Thought it's nice to be informed about it.

File: crates/validator/src/tests/ann.rs [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+        .expect("failed to run query")
+        .into_rows_result()
+        .expect("failed to get rows")
+        .rows()
+        .unwrap()
+        .collect::<Result<Vec<Row>, _>>()
+        .expect("failed to decode rows")
+}
+
+async fn ann_query_returns_expected_results(actors: TestActors) {
```
[the remark this answers is not in the document]
Comment (QuerthDP) [reply crates/validator/src/tests/ann.rs:48]: Would be appreciated if we decide on this convention. 
Anyway I'll be moving the utility functions to the `common.rs` file.

**Comment by QuerthDP** [comment]
Changelog:

- add common utils for creation of keyspaces, tables and indexes
- refactor existing tests to use the utils
- test limits at the edge values
